### PR TITLE
Implement `//arrai.info` to print arrai build information

### LINKED ISSUE #522: Expose arr.ai runtime info via //arrai.info
Suggested output format:

```arrai
@> //arrai.info
(
    version: 'DIRTY-v0.72.0',
    date: //time{2020-06-15T10:43:14Z},
    git: (
        commit: 'f633d75a049bca9343a442ada1d49577b676ff36',
        tags: {'v0.72.0'},
    ),
    go: (
        os: 'linux',
        arch: 'amd64',
        compiler: (
            version: 'go1.14.3',
            os: 'linux',
            arch: 'amd64',
        ),
    ),
)
```

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #522

Changes proposed in this pull request:
- Implements `//arrai.info` to print arr.ai build information

Current output is:
```
@> //arrai.info
(date: '2020-07-24T08:09:49Z', git: (commit: '52c577427456bee1754840cbe53455c0809a1e92', tags: {'v0.101.0'}), go: (arch: 'amd64', compiler: (arch: 'amd64', os: 'darwin', version: 'go1.14'), os: 'darwin'), version: 'DIRTY-52c577427456bee1754840cbe53455c0809a1e92 (from v0.101.0)')
@>  
```
And prettified output looks like:
```
@> //arrai.info
(
  date: '2020-07-24T08:09:49Z', 
  git: (
    commit: '52c577427456bee1754840cbe53455c0809a1e92', 
    tags: {'v0.101.0'}
  ), 
  go: (
    arch: 'amd64', 
    compiler: (
      arch: 'amd64', 
      os: 'darwin', 
      version: 'go1.14'
    ), 
    os: 'darwin'
  ), 
  version: 'DIRTY-52c577427456bee1754840cbe53455c0809a1e92 (from v0.101.0)'
)
@>  
```

Pretty output will be done by https://github.com/arr-ai/arrai/issues/171#issuecomment-662364599

Checklist:
- [x] Added related tests
- [] Made corresponding changes to the documentation

### CODE REVIEW

File: rel/prettify.go [hunk 1]
```diff
@@ -28,21 +28,25 @@ import (
 //		}
 //	)
 //}
-func FormatString(val rel.Value, indentsNum int) (string, error) {
+func PrettifyString(val Value, indentsNum int) (string, error) {
```
Comment (marcelocantos) [inline rel/prettify.go:31]: These changes don't belong in this PR.

File: syntax/std_runtime.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		rel.NewAttr("info", BuildInfo),
+	)
+}
+
+// GetBuildInfo returns arr.ai build information.
+func GetBuildInfo(version, date, fullCommit, tags, os, arch, goVersion string) rel.Value {
+	gitInfo := rel.NewTuple(
+		rel.NewAttr("commit", rel.NewString([]rune(fullCommit))),
+		// param tags has only one tag now.
+		rel.NewAttr("tags", rel.NewArray(rel.NewString([]rune(tags)))))
```
Comment (marcelocantos) [inline syntax/std_runtime.go:23]: Use a set, not an array.

File: syntax/std_runtime.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		rel.NewAttr("compiler", compiler))
+
+	info := rel.NewTuple(
+		rel.NewAttr("version", rel.NewString([]rune(version))),
+		rel.NewAttr("date", rel.NewString([]rune(date))),
+		rel.NewAttr("git", gitInfo),
+		rel.NewAttr("go", goInfo),
+	)
+
+	return info
```
Comment (marcelocantos) [inline syntax/std_runtime.go:51]: Just return expression directly.

File: syntax/std_runtime.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	goInfo := rel.NewTuple(
+		rel.NewAttr("os", rel.NewString([]rune(os))),
+		rel.NewAttr("arch", rel.NewString([]rune(arch))),
+		rel.NewAttr("compiler", compiler))
+
+	info := rel.NewTuple(
+		rel.NewAttr("version", rel.NewString([]rune(version))),
+		rel.NewAttr("date", rel.NewString([]rune(date))),
+		rel.NewAttr("git", gitInfo),
+		rel.NewAttr("go", goInfo),
```
Comment (marcelocantos) [inline syntax/std_runtime.go:48]: No need to build up the structure piecemeal. Create the nested structures inline.

File: syntax/std_runtime_test.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  go: (
+    arch: 'amd64',
+    compiler: (
+      arch: 'amd64',
+      os: 'darwin',
+      version: 'go1.14'
+    ),
+    os: 'darwin'
+  ),
+  version: 'DIRTY-v0.99.0'
```
Comment (marcelocantos) [inline syntax/std_runtime_test.go:31]:
```suggestion
    tags: ['v0.99.0'],
  ),
  go: (
    arch: 'amd64',
    compiler: (
      arch: 'amd64',
      os: 'darwin',
      version: 'go1.14',
    ),
    os: 'darwin',
  ),
  version: 'DIRTY-v0.99.0',
```

File: syntax/std_runtime.go (hunk 2 above)
Comment (ericzhang6222) [reply syntax/std_runtime.go:23]: Set will output `tags: {  'v0.99.0' }`, and the expected output is `tags: ['v0.72.0']`.

File: syntax/std_runtime.go (hunk 3 above)
Comment (ericzhang6222) [reply syntax/std_runtime.go:51]: If your meaning is coding something like:
```
expr, _ := EvaluateExpr(".", "(version: 'DIRTY-v0.101.0)', git: (commit: '52c577427456bee1', tags: ['v0.101.0'])")
val, err := expr.Eval(rel.Scope{})
```
It will output `(git: (commit: 52c577427456bee1, tags: [v0.101.0]), version: DIRTY-v0.101.0))`, the string value is not surrounded by quotes. That's why I built `tuple` by code. I am not sure if it is a bug,  can you confirm?

Comment (ericzhang6222) [reply syntax/std_runtime.go]: Found `rel.Repr(val rel.Value)` which will format values and add `'...'` to surround string value, so has changed code to:
```
// GetBuildInfo returns arr.ai build information.
func GetBuildInfo(version, date, fullCommit, tags, os, arch, goVersion string) rel.Value {
	expr, err := EvaluateExpr(".",
		fmt.Sprintf(buildInfoTemplate, version, date, fullCommit, tags, os, arch, goVersion, os, arch))
	if err != nil {
		return rel.NewString([]rune(err.Error()))
	}
	return expr
}
```

File: syntax/std_runtime.go (hunk 2 above)
Comment (ericzhang6222) [reply syntax/std_runtime.go]: Resolved as https://github.com/arr-ai/arrai/pull/521#discussion_r460511677

File: syntax/std_runtime.go [hunk 6]
```diff
@@ -1,14 +1,34 @@
...
+    go: (
+        os: '%s',
+        arch: '%s',
+        compiler: (
+            version: '%s',
+            os: '%s',
+            arch: '%s',
+        ),
+    ),
+)
```
Comment (marcelocantos) [inline syntax/std_runtime.go:29]: Pasting together arr.ai syntax with `fmt.Sprintf` is a code smell. The original code was preferable.

File: syntax/std_runtime.go (hunk 3 above)
Comment (marcelocantos) [reply syntax/std_runtime.go]: All I meant was, instead of:
```go
info := ....

return info
```
just write
```arrai
return ...
```

File: internal/shell/shell_cmd_test.go [hunk 7]
```diff
@@ -152,3 +152,14 @@ func assertEqualScope(t *testing.T, expected, actual rel.Scope) {
 		assert.True(t, v1.(rel.Value).Equal(v2))
 	}
 }
+
+// `//arrai.info` can't be constructed in this test as it involves package `main`.
+func removedVoidItems(scope rel.Scope) rel.Scope {
```
Comment (orlade-anz) [inline internal/shell/shell_cmd_test.go:157]: Nit: This name is slightly misleading: it's removing a very specific item (arrai.info), not all void items as the name suggests. Maybe `removeArraiInfo()`?

Also nit: s/removed/remove/
```suggestion
// removeArraiInfo removes `//arrai.info` from the scope as it can't be constructed in this test due to its lack of package `main`.
func removeArraiInfo(scope rel.Scope) rel.Scope {
```
